runtime/runtime_controller.h

```diff
@@ -558,6 +558,8 @@ class RuntimeController : public PlatformConfigurationClient {
     return context_.snapshot_delegate;
   }
 
+  std::weak_ptr<DartIsolate> GetRootIsolate() { return root_isolate_; }
```

Comment:
This should be removed.  The `RuntimeController` should not expose a mutable pointer to the `DartIsolate`.

`RuntimeController` historically did not have a `GetRootIsolate` API.  It looks like a const version of `GetRootIsolate` was added recently to support some unit tests.  If we keep that method, then it should be marked as being intended only for use by tests.

Reply:
done